pymc/aesaraf.py

```diff
@@ -899,11 +901,66 @@ def take_along_axis(arr, indices, axis=0):
     return arr[_make_along_axis_idx(arr_shape, indices, _axis)]
 
 
-def compile_rv_inplace(inputs, outputs, mode=None, **kwargs):
-    """Use ``aesara.function`` with the random_make_inplace optimization always enabled.
+@local_optimizer(tracks=[CheckParameterValue])
+def local_remove_check_parameter(fgraph, node):
+    """Rewrite that removes Aeppl's CheckParameterValue
 
-    Using this function ensures that compiled functions containing random
-    variables will produce new samples on each call.
+    This is used when compile_rv_inplace
+    """
+    if isinstance(node.op, CheckParameterValue):
+        return [node.inputs[0]]
+
+
+@local_optimizer(tracks=[CheckParameterValue])
+def local_check_parameter_to_ninf_switch(fgraph, node):
+    if isinstance(node.op, CheckParameterValue):
+        logp_expr, *logp_conds = node.inputs
+        if len(logp_conds) > 1:
+            logp_cond = at.all(logp_conds)
+        else:
+            (logp_cond,) = logp_conds
+        out = at.switch(logp_cond, logp_expr, -np.inf)
+        out.name = node.op.msg
+
+        if out.dtype != node.outputs[0].dtype:
+            out = at.cast(out, node.outputs[0].dtype)
+
+        return [out]
+
+
+for database in ("canonicalize", "stabilize", "specialize", "useless"):
+    aesara.compile.optdb[database].register(
+        "local_remove_check_parameter",
+        local_remove_check_parameter,
+        use_db_name_as_tag=False,
+    )
+
+    aesara.compile.optdb[database].register(
+        "local_check_parameter_to_ninf_switch",
+        local_check_parameter_to_ninf_switch,
+        use_db_name_as_tag=False,
+    )
```

Comment:
Why add these to all those databases?

Also, don't forget that you can set a priority that affects the order in which they're run.

Reply:
I just copied what I've seen elsewhere.

Can you give some recommendation on what is the minimum database(s) where these should be registered, as well as what the priority should be?

Reply:
You should be able to put those in just the pass named `"useless"` and have them removed before the other passes.  Check out `aesara.compile.mode` to see the passes and their ordering.

Otherwise, I would think that removing those checks should be a user-configurable option, and, if so, you might not want to include them by default.  Instead, an option could be set that includes those rewrites when graphs are compiled by PyMC.

Reply:
> Otherwise, I would think that removing those checks should be a user-configurable option, and, if so, you might not want to include them by default. Instead, an option could be set that includes those rewrites when graphs are compiled by PyMC (e.g. by constructing a Mode object for aesara.function that uses includes=...).

My understanding is that they are not running be default already due to the `use_db_name_as_tag=False` option. I am manually including them in `compile_pymc` helper function.

Reply:
> You should be able to put those in just the pass named "useless" and have them removed before the other passes. Check out aesara.compile.mode to see the passes and their ordering.

For some reason putting it in "useless" is not doing anything, but putting them in "canonicalize" seems to do the job.

Reply:
Looks like we need to refactor that `OptimizationDatabase` code; the whole `use_db_name_as_tag` thing, and the special way it works with only one DB implementation (i.e. `EquilibriumDB`) is not good.  Just the fact that there's a note about a specific subclass in the base class is a bad sign.

Anyway, `"useless"` doesn't use the same kind of underlying `OptimizationDatabase` types as `"canonicalize"`, so I'm guessing that the reason the latter works and the former doesn't is related to that.